packages/cli/test/dev/utils.js

```diff
@@ -98,7 +98,6 @@ function shouldSkip(name, versions) {
 
 function validateResponseHeaders(res, podId) {
   if (res.status < 500) {
-    expect(res.headers.get('server')).toEqual('Vercel');
```

Comment:
This assertion is no longer always true. I think we can remove it.

Reply:
I believe it's true in production. I think we should look into maintaining that behavior for dev.

Reply:
It is true for a lot of cases but I believe the case that needs to be looked into is when vc dev does a `proxyPass()`, since then the destination headers will take precedence over the ones already set.

I'm guessing that this validation logic was not being used in tests that do a proxy pass until this PR.

Reply:
Perhaps we should add this line back and remove the `validateResponseHeaders()` from the proxy pass test.

Reply:
When the proxy proxies to another origin, it was setting the response header from that target.

I figured out how to override this. Updated [`ae53c87` (#8457)](https://github.com/vercel/vercel/pull/8457/commits/ae53c87890768b8b760c2dd690da5b38aacbd3cd)